gufe/chemicalstate.py

```diff
@@ -60,11 +60,27 @@ def __init__(
         else:
             self._box_vectors = box_vectors
 
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        if self.identifier != other.identifier:
```

Comment:
I think the only reason to have all these early exits is if we think equality checking is likely to be a significant cost. In that case, I'd go with the direct `._identifier`, rather than invoking he overhead of the property. (And similar with other things: `_box_vectors`, `_components`.)